### ISSUE TITLE
Softmotion - Homing Position Mode Enhancements

### ISSUE THREAD

**Issue by SimonRPSM** [LabVIEW-DCAF/MotionModule#4]
Currently the mode uses the Softmotion settings (Start Search Forward, Stop on FWD Edge, Approach FWD).

Write code to go from these to the DS402 32 possible reference moves.

**Comment by Karl-G1** [comment]
@SimonRPSM 

Of the 31 supported Homing Modes in DS402, LabVIEW SoftMotion supports 13. By adding a Find Reference: Index move after Modes 17-18 and 23-30, you can easily implement Modes 1-2 and 7-14 in this module. Modes 3-6 and 19-22 would not be terribly difficult to implement if needed, as you just need to know the current status of the home switch.

Keep in mind however that most EtherCAT drives do not support all 13 of SoftMotion's homing modes, and that SDI Plug-ins work around this by commanding supported modes with the closest parameters. For first-party SoftMotion hardware, 12 homing modes are supported (all except 35: absolute).

![image](https://user-images.githubusercontent.com/31290917/29681628-ea7d70b0-88cd-11e7-86e8-801e6f792558.png)

**Comment by SimonRPSM** [comment]
I've been thinking about how to implement the 2 moves and I am not sure whether it should be the motion module that spins of the second (index) move. This would be easy to implement in the Sequencer, but a bit more tricky to set up in the module itself. I will be shelving this one for now and stick to only implementing (in the module) the ones that Softmotion natively supports.

**Comment by Karl-G1** [comment]
I agree; I wouldn't worry about the index move modes for now. I would handle it in the motion module by essentially having that 'move' be two separate Find Reference Moves.

**Comment by SimonRPSM** [comment]
Sounds good, @KarlG-Motion 

For now I'll look into how I think I'll create a small converter to go from DS402 Stop -> My Cluster. That should be enough at this time - then I can look into the double-move thing if necessary.

**Comment by aggieslaughter** [comment]
Can we support Absolute moves as we discussed at my desk (the change I made in the code with you here) by just calling the absolute move and passing no parameters except the homing type?

**Comment by Karl-G1** [comment]
Would be easy enough to test, but I believe that SoftMotion will not check for valid inputs on an Absolute homing move considering nothing else should be required by the drive.
